api/app/jobs/fetch_video_thumbnails.rb

```diff
@@ -0,0 +1,10 @@
+# Job to fetch youtube and vimeo thumbnails in the background.
+class FetchVideoThumbnails < ApplicationJob
+  queue_as :default
+
+  def perform(video_id)
+    video = Resource.find(video_id)
+    return unless video
+    VideoThumbnail::Fetcher.new.fetch(video)
```

Comment:
As with the job, the class is too narrowly focused. Thumbnail::Fetcher instead of VideoThumbnail::Fetcher